### PR TITLE
feat(codepipeline): `executionMode` property for Pipeline

### LINKED ISSUE #29147: aws-codepipeline: add `executionMode` property to Pipeline
### Describe the feature

CloudFormation already supports `executionMode`, which can also be supported by CDK.

https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-resource-codepipeline-pipeline.html#cfn-codepipeline-pipeline-executionmode

### Use Case

We would be good to add a new parameter for execution mode.

### Proposed Solution

Add an `executionMode` parameter to the `PipelineProps` interface.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.128.0

### Environment details (OS name and version, etc.)

MacOS 13.0

### PR DESCRIPTION
**Pull request by go-to-k**
### Issue # (if applicable)

Closes #29147.

### Reason for this change

We would be good to add a new parameter for execution mode.

see:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codepipeline-pipeline.html#cfn-codepipeline-pipeline-executionmode

https://aws.amazon.com/about-aws/whats-new/2024/02/codepipeline-trigger-filters-execution-modes

### Description of changes

Add an `executionMode` parameter to the `PipelineProps` interface.

### Description of how you validated changes

Both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts [hunk 1]
```diff
@@ -495,6 +521,14 @@ export class Pipeline extends PipelineBase {
...
     this.pipelineType = props.pipelineType ?? PipelineType.V1;
 
+    if (
+      props.executionMode
+      && [ExecutionMode.QUEUED, ExecutionMode.PARALLEL].includes(props.executionMode)
+      && this.pipelineType !== PipelineType.V2
+    ) {
+      throw new Error(`${props.executionMode} execution mode can only be used with V2 pipelines, \`PipelineType.V2\` must be specified for \`pipelineType\``);
+    }
+
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts:531]: I have confirmed that the error occurs when QUEUED or PARALLEL mode is specified in the AWS console.

```
QUEUED or PARALLEL mode can only be used with V2 pipelines
```

File: packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts [hunk 2]
```diff
@@ -109,6 +109,24 @@ export enum PipelineType {
   V2 = 'V2',
 }
 
+/**
+ * Execution mode.
+ */
+export enum ExecutionMode {
```
Comment (jumic) [inline packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts:115]: asdf

File: packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts [hunk 3]
```diff
@@ -109,6 +109,24 @@ export enum PipelineType {
...
+  /**
+   * QUEUED mode
```
Comment (jumic) [inline packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts:117]:
```suggestion
   * QUEUED mode: Executions are processed one by one in the order that they are queued. This requires pipeline type V2.
```

File: packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts [hunk 4]
```diff
@@ -109,6 +109,24 @@ export enum PipelineType {
...
+   */
+  QUEUED = 'QUEUED',
+  /**
+   * SUPERSEDED mode
```
Comment (jumic) [inline packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts:121]:
```suggestion
   * SUPERSEDED mode: A more recent execution can overtake an older one. This is the default.
```

File: packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts [hunk 5]
```diff
@@ -109,6 +109,24 @@ export enum PipelineType {
...
+   */
+  SUPERSEDED = 'SUPERSEDED',
+  /**
+   * PARALLEL mode
```
Comment (jumic) [inline packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts:125]:
```suggestion
   * PARALLEL mode: In PARALLEL mode, executions run simultaneously and independently of one another. Executions don't wait for other runs to complete before starting or finishing. This requires pipeline type V2.
```

**Review by jumic — CHANGES_REQUESTED** [review]
From my point of view, the PR already looks good. I would suggest adding some descriptions for the execution modes. (Copied from https://docs.aws.amazon.com/codepipeline/latest/userguide/concepts-how-it-works.html)

**Comment by go-to-k** [comment]
@jumic 

Thanks for your review. I modified them, could you please check? (The format of line breaks, etc. has been slightly changed.)

**Review by jumic — APPROVED** [review]
Thanks. LGTM! 👍

**Review by gracelu0 — DISMISSED** [review]
LGTM, thank you for your contribution! :)

**Comment by go-to-k** [comment]
@gracelu0 

Thanks for your approval.
The merge of the main branch caused a difference in the integ test, so I modified the snapshot again.
Could you please approve it again?

```
  CHANGED    aws-codepipeline-actions/test/integ.pipeline-type-v2 3.224s
      Resources
[~] AWS::Lambda::Function CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F 
 └─ [~] Code
     └─ [~] .S3Key:
         ├─ [-] 2ec8ad9e91dcd6e7ad6a5c84ffc6c9c05c408aca3b26ceb2816d81043e6c4dc3.zip
         └─ [+] 54e6f07bf3aea46d100719f3b21fe0fb45ae789ee372d42d8cdfb97e21aba96c.zip
```

**Review by gracelu0 — APPROVED** [review]

**Pull request merged**